### PR TITLE
Add RpcFieldsGenerator for parsing rpc.* tags

### PR DESCRIPTION
**Pull request by mohit-a21**
rpc.* tags have been defined for Otel. Added Protocol generator
for such tags. The custom rpc.* tags defined gets translated to
relevant rpc protocol such as grpc, java_rmi, etc. Since we support
grpc only, for now, added support for translating of tags to grpc
event field

Dependent on https://github.com/hypertrace/data-model/pull/9

### CODE REVIEW

File: raw-span-constants/src/main/proto/org/hypertrace/core/span/constants/v1/envoy_attributes.proto [hunk 1]
```diff
@@ -17,15 +17,16 @@ enum Envoy {
...
-  ENVOY_HTTP_PROTOCOL = 13 [(string_value) = "http.protocol"];
-  ENVOY_NODE_ID = 14 [(string_value) = "node_id"];
-  ENVOY_RESPONSE_FLAGS = 15 [(string_value) = "response_flags"];
-  ENVOY_RETRY_COUNT = 16 [(string_value) = "retry.count"];
-  ENVOY_STATUS = 17 [(string_value) = "status"];
-  ENVOY_UPSTREAM_ADDRESS = 18 [(string_value) = "upstream_address"];
-  ENVOY_UPSTREAM_CLUSTER = 19 [(string_value) = "upstream_cluster"];
-  ENVOY_USER_AGENT = 20 [(string_value) = "user_agent"];
-  ENVOY_ZONE = 21 [(string_value) = "zone"];
+  ENVOY_GRPC_AUTHORITY = 11 [(string_value) = "grpc.authority"];
```
Comment (tim-mwangi) [inline raw-span-constants/src/main/proto/org/hypertrace/core/span/constants/v1/envoy_attributes.proto:20]: Why did you change the numbering of these proto fields?

Comment (mohit-a21) [reply raw-span-constants/src/main/proto/org/hypertrace/core/span/constants/v1/envoy_attributes.proto:20]: I wanted to keep the grpc.* tags together.

Comment (skjindal93) [reply raw-span-constants/src/main/proto/org/hypertrace/core/span/constants/v1/envoy_attributes.proto:20]: That would be a breaking change!

https://developers.google.com/protocol-buffers/docs/proto#assigning_field_numbers
> These numbers are used to identify your fields in the message binary format, and should not be changed once your message type is in use

Comment (mohit-a21) [reply raw-span-constants/src/main/proto/org/hypertrace/core/span/constants/v1/envoy_attributes.proto:20]: Yeah. But I don't think we use proto rawconstant values as such. We use the string values of enums that are defined here. Are we storing the numeric values somewhere?

File: span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/RpcFieldsGenerator.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      RawSpanConstants.getValue(RPC_REQUEST_METADATA) + DOT;
+  private static final String RESPONSE_METADATA_PREFIX =
+      RawSpanConstants.getValue(RPC_RESPONSE_METADATA) + DOT;
+  private static Logger LOGGER = LoggerFactory.getLogger(RpcFieldsGenerator.class);
+  private static Map<String, FieldGenerator<Rpc.Builder>> fieldGeneratorMap = initializeFieldGenerators();
+
+  GrpcFieldsGenerator grpcFieldsGenerator;
+
+  public RpcFieldsGenerator(GrpcFieldsGenerator grpcFieldsGenerator) {
+    this.grpcFieldsGenerator = grpcFieldsGenerator;
```
Comment (tim-mwangi) [inline span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/RpcFieldsGenerator.java:37]: why does this need to take an internal field generator?

Comment (mohit-a21) [reply span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/RpcFieldsGenerator.java:37]: rpc.* tags will get translated to relevant grpc or other rpc protocol fields based on the value of tag `rpc.system`. For eg: rpc.request.metadata.* tags get processed by grpcFieldsGenerator after validating rpc.system = grpc.

File: raw-span-constants/src/main/proto/org/hypertrace/core/span/constants/v1/envoy_attributes.proto (hunk 1 above)
Comment (buchi-busireddy) [reply raw-span-constants/src/main/proto/org/hypertrace/core/span/constants/v1/envoy_attributes.proto:20]: @mohit-a21 We have decided *not* to add any more proto constants. These should be Java string constants so why not start doing that? For a while it might be little inconsistent but we'll have to take that step anyways.

File: raw-span-constants/src/main/proto/org/hypertrace/core/span/constants/v1/opentelemetry.proto [hunk 3]
```diff
@@ -0,0 +1,31 @@
+syntax = "proto3";
+option java_multiple_files = true;
+package org.hypertrace.core.span.constants.v1;
+
+import "org/hypertrace/core/span/constants/v1/enum_extension.proto";
+
+// This file contains constants related to open telemetry tracer.
+// Written as per
+enum OTELSpanTag {
```
Comment (buchi-busireddy) [inline raw-span-constants/src/main/proto/org/hypertrace/core/span/constants/v1/opentelemetry.proto:9]: Please add these as Java enum constants and keep the usage private to this repo.

File: raw-span-constants/src/main/proto/org/hypertrace/core/span/constants/v1/span_attribute.proto [hunk 4]
```diff
@@ -92,6 +92,23 @@ enum Grpc {
     GRPC_METHOD = 10 [(string_value) = "grpc.method"];
 }
 
+enum Rpc {
```
Comment (buchi-busireddy) [inline raw-span-constants/src/main/proto/org/hypertrace/core/span/constants/v1/span_attribute.proto:95]: Same here. Every constant here should be local to this service and all these should be first class fields on the span right?

File: span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/RpcFieldsGenerator.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    JaegerSpanInternalModel.KeyValue keyValue = tagsMap.get(OTEL_SPAN_TAG_RPC_SYSTEM_ATTR);
+    if (keyValue != null) {
+      String val = ValueConverter.getString(keyValue);
+      return val;
+    }
+    return null;
+  }
+
+  private boolean isRpcSystemGrpc(String rpcSystem) {
+    if (StringUtils.isNotBlank(rpcSystem) && StringUtils.equals(rpcSystem, OTEL_RPC_SYSTEM_GRPC_VAL)) {
```
Comment (buchi-busireddy) [inline span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/RpcFieldsGenerator.java:112]: Return with condition directly instead of `if` statement?

File: raw-span-constants/src/main/proto/org/hypertrace/core/span/constants/v1/span_attribute.proto (hunk 4 above)
Comment (mohit-a21) [reply raw-span-constants/src/main/proto/org/hypertrace/core/span/constants/v1/span_attribute.proto:95]: All of these are named fields in Grpc event.

File: span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/RpcFieldsGenerator.java (hunk 5 above)
Comment (mohit-a21) [reply span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/RpcFieldsGenerator.java]: Done

File: raw-span-constants/src/main/proto/org/hypertrace/core/span/constants/v1/opentelemetry.proto (hunk 3 above)
Comment (mohit-a21) [reply raw-span-constants/src/main/proto/org/hypertrace/core/span/constants/v1/opentelemetry.proto]: Done

File: span-normalizer-constants/build.gradle.kts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+plugins {
+  `java-library`
+  jacoco
+  id("org.hypertrace.jacoco-report-plugin")
+}
+
+dependencies {
+  testImplementation("junit:junit:4.11")
```
Comment (tim-mwangi) [inline span-normalizer-constants/build.gradle.kts:8]: nope. jupiter junit5

Comment (mohit-a21) [reply span-normalizer-constants/build.gradle.kts]: Updated.

**Review by tim-mwangi — APPROVED** [review]
